File: NCore/Commands/Commandnmc.cs

using System.Threading.Tasks;

namespace NCore
{
    public class Commandnmc : Command
    {
        public Commandnmc() : base("nmc", "Выключает проверку движения [ТОЛЬКО ИГРОК]", "nmc")
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            if (!sender.IsPlayer)
            {
                await sender.SendMessage("Только для игрока.");
                return true;
            }

            if (sender.GetAdmin())
            {
                NetworkPlayer p = (NetworkPlayer)sender;
                p.DisableMovementCheck = !p.DisableMovementCheck;
                if (p.DisableMovementCheck)
                {
                    await sender.SendMessage("Отключена проверка передвижения.");
                }

                return true;
            }

            return false;
        }
    }
}

File: NetcraftServer/Commands/Commandnmc.vb
Imports NetcraftServer

Public Class Commandnmc
    Inherits Command
    Sub New()
        MyBase.New("nmc", "Выключает проверку движения [ТОЛЬКО ИГРОК]", "nmc")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.IsPlayer Then
            sender.SendMessage("Только для игрока.")
            Return True
        End If
        If sender.GetAdmin Then
            Dim p As NetworkPlayer = sender
            p.DisableMovementCheck = Not p.DisableMovementCheck
            If p.DisableMovementCheck Then
                sender.SendMessage("Отключена проверка передвижения.")
            End If
            Return True
        End If
        Return False
    End Function
End Class
